### PR TITLE
feat(gatsby): add babel-lodash plugin to reduce lodash filesize

### PR DESCRIPTION
**Pull request by wardpeet**
# Description

I moved bundle optimization aliases to a different plugin so it's abstracted away and easier to see what optimizations we use.

The real feature is reducing lodash's footprint by using [babel-plugin-lodash](https://npm.im/babel-plugin-lodash). This will convert user's code to only use the imported features and not the whole library.

It will improve gatsbyjs.com performance.

You can test it with `yarn add --registry=https://registry.wardpeet.dev gatsby@lodash`

## Example
```
import { get } from "lodash"
```

Size before: "full lodash loaded 70.43KB"
Size after: "lodash/get loaded 6.64KB"

### CODE REVIEW

**Comment by DSchau** [comment]
@wardpeet if this is in Core, does this invalidate the usefulness of [this plugin](https://www.gatsbyjs.com/plugins/gatsby-plugin-lodash/)? If so -- should we deprecate it?

**Comment by wardpeet** [comment]
TIL that we had that plugin. That plugin still has `lodash-webpack-plugin` but I don't think there is much win in having that one enabled by default or having it at all.

We probably should after this got published, good call.

**Review by pvdz — DISMISSED** [review]
Sounds good to me :D

File: yarn.lock [hunk 1]
```diff
@@ -5421,7 +5421,7 @@ babel-plugin-jest-hoist@^24.9.0:
   dependencies:
     "@types/babel__traverse" "^7.0.6"
 
-babel-plugin-lodash@^3.3.4:
+babel-plugin-lodash@3.3.4, babel-plugin-lodash@^3.3.4:
```
Comment (pvdz) [inline yarn.lock:5424]: Something in our stack already depends on this? Hope it's not us :)

**Review by pvdz — APPROVED** [review]
:+1:

File: yarn.lock (hunk 1 above)
Comment (wardpeet) [reply yarn.lock]: We have a plugin that does this https://www.gatsbyjs.com/plugins/gatsby-plugin-lodash/ so I'll deprecate that later.

**Review by laurieontech — APPROVED** [review]

**Comment by Blumed** [comment]
@wardpeet what is the best way to know if this worked in my current codebase? This change I would assume only happens during production builds. Thinking out loud, it would be nice to have a saving notification print during build like many other tree shaking packages.

**Comment by wardpeet** [comment]
it only runs during production, so you'll have to do a `gatsby build` and check the sourcefiles. I recommend using https://www.gatsbyjs.com/plugins/gatsby-plugin-webpack-bundle-analyser-v2/

**Comment by lsanwick** [comment]
One issue with this change is that breaks any project currently using Lodash's `chain` method, as the plugin doesn't support the method.

I wish this remained a plugin I could opt into rather than a core part of the Gatsby dependencies.

**Comment by wardpeet** [comment]
@isanwick you you create an issue with reproduction please?

**Pull request merged**